CMakeLists.txt

```diff
@@ -113,9 +97,33 @@ if(${CI_BUILD})
     endif()
 endif()
 
+# Link Time Optimization (LTO)
+cmake_policy(SET CMP0069 NEW)
+option(ENABLE_LTO "Set to ON to build Hyrise with enabled coverage checking. Default: OFF" OFF)
+if (${ENABLE_LTO})
+    if(NOT ${CMAKE_VERSION} VERSION_LESS "3.9.0")
+        cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
+        include(CheckIPOSupported)
+        check_ipo_supported(RESULT ipo_supported OUTPUT ipo_output)
+        if(ipo_supported)
+            set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+            add_definitions(-DWITH_LTO)
+            message(STATUS "Building with Link-Time Optimization")
+        else()
+            message(WARNING "LTO not supported: ${ipo_output}")
+        endif()
+    endif()
+endif()
+
 # Include sub-CMakeLists.txt
 add_subdirectory(third_party/ EXCLUDE_FROM_ALL)
+
+# Some third-party libs don't support LTO (if enabled above):
```

Comment:
Why do we need to handle this? Just asking because I'm not getting any Warnings or Errors regarding LTO in these libs.

Reply:
```
CMake Warning (dev) at third_party/googletest/googletest/cmake/internal_utils.cmake:149 (add_library):
  Policy CMP0069 is not set: INTERPROCEDURAL_OPTIMIZATION is enforced when
  enabled.  Run "cmake --help-policy CMP0069" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  INTERPROCEDURAL_OPTIMIZATION property will be ignored for target 'gtest'.
```

Are you building with ENABLE_LTO?

Reply:
Ah I was testing this one on the master version since we did not handle it there, and there it didn't complain. But yes, on your branch it complains for me too without these.

Reply:
Well yes - that's because the master never did any LTO...